homeassistant/helpers/deprecation.py

```diff
@@ -194,10 +213,56 @@ def _print_deprecation_warning(
         else:
             logger.warning(
                 "%s was %s from %s, this is a deprecated %s%s. Use %s instead",
-                obj.__name__,
+                obj_name,
                 verb,
                 integration_frame.integration,
                 description,
                 breaks_in,
                 replacement,
             )
+
+
+DeprecatedConstant = namedtuple(
+    "DeprecatedConstant", ("value", "replacement", "breaks_in_ha_version")
+)
+DeprecatedConstantEnum = namedtuple(
+    "DeprecatedConstantEnum", ("enum", "breaks_in_ha_version")
+)
+
+
+def check_if_deprecated_constant(name: str, module_globals: dict[str, Any]) -> Any:
+    """Check if the not found name is a deprecated constant.
+
+    If it is, print a deprecation warning and return the value of the constant.
+    Otherwise raise AttributeError.
+    """
+    module_name = module_globals.get("__name__")
+    logger = logging.getLogger(module_name)
+    if (deprecated_const := module_globals.get(f"_DEPRECATED_{name}")) is None:
+        raise AttributeError(f"Module {module_name!r} has no attribute {name!r}")
+    if isinstance(deprecated_const, DeprecatedConstant):
+        value = deprecated_const.value
+        replacement = deprecated_const.replacement
+        breaks_in_ha_version = deprecated_const.breaks_in_ha_version
+    elif isinstance(deprecated_const, DeprecatedConstantEnum):
+        value = deprecated_const.enum.value
+        replacement = (
+            f"{deprecated_const.enum.__class__.__name__}.{deprecated_const.enum.name}"
+        )
+        breaks_in_ha_version = deprecated_const.breaks_in_ha_version
+    else:
+        logger.debug(
+            "Invalid value for deprecated constant %s, should be DeprecatedConstant or DeprecatedConstantEnum",
+            name,
+        )
+        raise AttributeError(f"Module {module_name!r} has no attribute {name!r}")  # noqa: TRY004
```

Comment:
Shouldn't we raise a different error message than above?

Reply:
Maybe `"Module {module_name!r} has no attribute {name!r} and the deprecated constant has the wrong format"`?

Reply:
This case is not for a missing attribute. It's the case for incorrect type.